python/sdist/amici/de_export.py

```diff
@@ -1638,6 +1646,132 @@ def free_symbols(self) -> set[sp.Basic]:
             )
         )
 
+    def static_indices(self, name: str) -> list[int]:
+        """
+        Returns the indices of static expressions in the given model entity.
+
+        Static expressions are those that do not depend on time,
+        neither directly nor indirectly.
+
+        :param name: Name of the model entity.
+        :return: List of indices of static expressions.
+        """
+        # already computed?
+        if (res := self._static_indices.get(name)) is not None:
+            return res
+
+        if name == "w":
+            dwdx = self.eq("dwdx")
+            dwdw = self.eq("dwdw")
+            w = self.eq("w")
+
+            # check for indirect state and time dependency
+            # to avoid lengthy symbolic computations,
+            # we only check if we have any non-zeros in hierarchy,
+            # we currently neglect the case where different hierarchy
+            # levels may cancel out. treating a static expression as
+            # dynamic in such rare cases shouldn't be a problem.
+            nonzero_dwdx = np.asarray(
```

Comment:
rename `nonzero_dwdx` to something like `dynamic_dependency` and explicitly state in the comments that we are looking for direct dependency (via `t`) and indirect dependency (via `x`).